server/lib/topic.js

```diff
@@ -61,13 +63,15 @@ const channels = {
 // writes events into the event database table
 function databaseLogger(data) {
 
-  let record = {
+  const record = {
     timestamp: new Date(data.timestamp),
     user_id: data.user_id,
     channel: data.channel,
+    entity : data.entity.toUpperCase(),
```

Comment:
No having it be required is no problem! It just might be a good idea to formalise what is needed when passing in data to publish method. 

In the future getting the error 'Publishing an event requires an entity' vs. 'cannot call function toUpperCase of undefined' might save some time - this isn't a user facing issue though so just a comment.